src/Controller/PositionController.php

```diff
@@ -0,0 +1,372 @@
+                $multisite[]=$this->config("Multisites-site{$i}");
+                $selectedSites[]=$selected;
+            }
+        }
+
+        $this->templateParams(array(
+            'id' => $id,
+            'nom' => $nom,
+            'etage' => $etage,
+            'groupe' => $groupe,
+            'group-id'=>$groupe_id,
+            'categories' =>$categories,
+            'site' => $site,
+            'activites' => $activites,
+            'obligatoire'=> $obligatoire,
+            'renfort' => $renfort,
+            'stat1' => $stat1,
+            'stat2' => $stat2,
+            'bloq1' => $bloq1,
+            'bloq2' => $bloq2,
+            'etages' => $etages,
+            'groupes' => $groupes,
+            'nbSites'=> $nbSites,
+            'multisite'=>$multisite,
+            'actList' => $actList,
+            'categoriesList'=>$categories_list,
+            'usedGroups'=> $groupes_utilises,
+            'selectedSites' => $selectedSites,
+            'CSRFToken' => $GLOBALS['CSRFSession']
+            )
+        );
+        return $this->output('position/edit.html.twig');
+    }
+
+    /**
+     * @Route("/position", name="position.save", methods={"POST"})
+     */
+    public function save(Request $request, Session $session)
+    {
+        $CSRFToken = $request->get('CSRFToken');
+        $nom=$request->get('nom');
+        $id = $request->get('id');
+
+
+        if (!$nom) {
+            $session->getFlashBag()->add('error',"Le nom est obligatoire");
+            if(!$id){
+                return $this->redirectToRoute('position.add');
+            } else {
+                return $this->redirectToRoute('position.edit', array('id' => $id));
+            }
+        }else{
+
+            $activites = json_encode($request->get('activites'));
+            $categories = json_encode($request->get('categories'));
+            $site = $request->get('site');
+            $bloquant= $request->get('bloquant');
+            $statistiques= $request->get('statistiques');
+            $etage = $request->get('etage');
+            $groupe = $request->get('groupe');
+            $groupe_id = $request->get('group-id');
+            $obligatoire= $request->get('obligatoire');
+            $site=$site?$site:1;
+
+            if (!$id){
+                    $position = new Position;
+                    $position->nom($nom);
+                    $position->activites($activites);
+                    $position->categories($categories);
+                    $position->bloquant($bloquant);
+                    $position->statistiques($statistiques);
+                    $position->etage($etage);
+                    $position->groupe($groupe);
+                    $position->groupe_id($groupe_id);
+                    $position->obligatoire($obligatoire);
+                    $position->site(site);
+                    $this->entityManager->persist($position);
+
+                    if (isset($error)) {
```

Comment:
D'où sort la variable "error" ?
Il faudrait avoir un retour de entityManager->persist.

Reply:
Je voulais stocker une erreur dans la variable error. Je vais regarder s'il y a un équivalent du try en C++

Reply:
Ou voir si entityManager ne renvoie pas un status. Il suffit peut être simplement de mettre
if( $this->entityManager->persist($position) ) {
   // Gestion du succès
} else {
   // Gestion de l'erreur
}
A tester